django/core/checks/translation.py

```diff
@@ -24,6 +29,11 @@
     id='translation.E004',
 )
 
+W005 = Warning(
+    'Inconsistent plural forms across catalogs for language {!r} (unmerged catalogs).',
+    id='translation.W005',
```

Comment:
That's the warning at a system check level. You will see that when you have catalogs with inconsistent plural forms when the checks are run.

The system check level warning is needed to inform the user about the inconsistency in all the supported languages (set by the LANGUAGES settings). Because of how i18n works in Django, every translation is lazily-loaded and cached, that makes loading only the default language on start, the rest is of the load is done when something request it.

The system check resets the translation cache and triggers the loading all the languages supported to see if there are any run-time warnings (also needed when adding them "on-the-fly"). After that, it resets the cache again and leaves the situation as it was before.

Run-time warnings are done when merging the catalogs, in [django.utils.translation.trans_real ](https://github.com/django/django/pull/12280/files?file-filters%5B%5D=.py&file-filters%5B%5D=.txt&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile#diff-5682903b7aed1649cdb7095331e90d0dR197)